core/src/main/scala/org/apache/spark/deploy/history/FsHistoryProvider.scala

```diff
@@ -557,7 +560,8 @@ private[history] class FsHistoryProvider(conf: SparkConf, clock: Clock)
    */
   private def replay(
       eventLog: FileStatus,
-      bus: ReplayListenerBus): Option[FsApplicationAttemptInfo] = {
+      bus: ReplayListenerBus,
+      eventsFilter: ReplayEventsFilter = SELECT_ALL_FILTER): Option[FsApplicationAttemptInfo] = {
```

Comment:
This helper replay method will need to be supplied a listener bus externally because the two call sites need very different sets of listeners on the bus. Only thing it looked we could do was enrich the return value from the helper to prevent the second callsite (getAppUI() ) from having to add and parse from an additional listener of its own. I've pushed the updated commit.